src/lib/icli/index.js

```diff
@@ -0,0 +1,354 @@
+ * @return {function} - The function that must be passed to cmd.action()
+ */
+function getAction(parameters, executeCommand, commanderActionHook, inquirerPromptHook) {
+  return function () {
+    // Hook that allows to tranform the result of the commander parsing, before converting it into parameter values
+    const args = commanderActionHook ? commanderActionHook.apply(this, arguments) : arguments;
+    const validations = _.reduce(parameters, (result, parameter) => {
+      if (parameter.validate) {
+        result[parameter.name] = parameter.validate;
+      }
+      return result;
+    }, {});
+    const commandParameterValues = processCliArgs(args, validations);
+
+    // If the cli arguments are correct, we can prepare the questions for the interactive prompt
+    // Launch the interactive prompt
+    return prompt.prompt(parametersToQuestions(parameters, commandParameterValues))
+    .then(answers => {
+      // Hook that allows to tranform the result of the inquirer prompt, before converting it into parameter values
+      if (inquirerPromptHook) {
+        return inquirerPromptHook(answers, commandParameterValues);
+      }
+      // Once we have all parameter values from the command and from the questions, we can execute the command
+      _.forEach(answers, (value, key) => {
+        if (!commandParameterValues[key] && answers[key]) {
+          commandParameterValues[key] = answers[key];
+        }
+      });
+      executeCommand(commandParameterValues);
+    });
+  };
+}
+
+/**
+ * Transform the parameters of commander action() callback into a list of parameter values
+ * and apply validations
+ * @param  {Object} cliArgs - "arguments" object passed to the method action()
+ * @param  {Object} validators - map of parameterKey / validation function
+ * @return void
+ */
+function processCliArgs(cliArgs, validations) {
+  // Initialize an object that will contain the final parameters (cli + prompt)
+  let parameters = cliArgsToParameters(cliArgs);
+  validations = validations || [];
+
+  // We verify that arguments provided in the command are correct
+  // For example, check if the provided api-identifier does really exist
+  // If an argument is a comma separated list, we could also transform it into an Array here
+  const validationResult = validateParameters(parameters, validations);
+  if (validationResult !== true) {
+    console.log('\n  ' + icli.format.ko('error') + ':\n    ' + validationResult.join('\n    ') + '\n');
+    process.exit(1);
+  }
+  return parameters;
+}
+
+/**
+ * Verify and transform cli args into parameters
+ * @param  {Object} cliArgs - arguments and options that have been passed to the cli
+ * @return {Object} - parameter values
+ */
+function cliArgsToParameters(cliArgs) {
+  // Initialize an object that will contain the final parameters (cli + prompt)
+  const parameters = {};
+
+  let options = Array.prototype.pop.call(cliArgs);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [`options` is never modified, use `const` instead.](https://www.codacy.com/app/lager/lager/file/2779185918/issues/source?bid=3360491&fileBranchId=3365065#l274)